pkg/template/processor.go

```diff
@@ -0,0 +1,156 @@
+package template
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sort"
+	"time"
+
+	projectv1 "github.com/openshift/api/project/v1"
+	templatev1 "github.com/openshift/api/template/v1"
+	"github.com/openshift/library-go/pkg/template/generator"
+	"github.com/openshift/library-go/pkg/template/templateprocessing"
+	errs "github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Processor struct {
+	cl           client.Client
+	scheme       *runtime.Scheme
+	codecFactory serializer.CodecFactory
+}
+
+func NewProcessor(cl client.Client, scheme *runtime.Scheme) *Processor {
+	return &Processor{cl: cl, scheme: scheme, codecFactory: serializer.NewCodecFactory(scheme)}
+}
+
+func (p Processor) ProcessAndApply(tmplContent []byte, values map[string]string) error {
```

Comment:
should we add some documentation on our public funcs?

Reply:
sure, let me do that

Reply:
done in e4cc320